### ISSUE TITLE
Sort by date of next episode

### ISSUE THREAD

**Issue by Scumpeter** [ltGuillaume/DroidShows#106]
When sorting by unseen, the shows should show the shows sorted by the date of the next unseen episode (the same value that is used for the "next episode" field in the overview).
Instead the shows are seemingly in random order:
![Screenshot_20211226-155554](https://user-images.githubusercontent.com/293778/147411923-851644ba-5c42-435e-9dfd-1e73a903fe19.png)

Tested with 7.11.2 from Fdroid on Android 12.

**Comment by ltguillaume** [comment]
Sort by unseen episodes == sort by *amount* of unseen episodes.  It's meant to help you decide which shows have enough episodes "available" for a binge night 🙈 I get the confusion, though, I even seem to have described the old behavior of the feature in readme, will fix that.

Nice list by the way 👍

**Comment by Scumpeter** [comment]
Ah. Thanks for the clarification. Should I create a feature request for an additional sorting order? It would really help me with watching Stargate in airing order 😉

**Comment by ltguillaume** [comment]
Ah you mean due to cross-overs?

**Comment by Scumpeter** [comment]
For example. Also to understand the possible references in one show to another show.

**Comment by ltguillaume** [comment]
Yeah, I'll have to restructure the database for that, but I'll look into it.

**Comment by ltguillaume** [comment]
I suppose this is _sorted_ (see what I did there? 😉) in @warren-bank's build, right?

**Comment by warren-bank** [comment]
actually.. I didn't do what @Scumpeter has requested..

when a user has several unseen episodes.. there's a first (oldest) and last (newest).

this request (I believe) is to sort all shows by the date (ascending) of each first (oldest) episode.

what I implemented in my fork was an option to sort all shows by the date (descending) of each last (newest) episode.

to be perfectly honest, I don't see any value in sorting by oldest.
the reason that I added the option to sort by newest is to see which shows have most recently aired new (unseen) episodes.

**Comment by ltguillaume** [comment]
Whoops, I knew I had to double-check this before posting something about it 😛

**Comment by warren-bank** [comment]
no worries..

strictly speaking, adding another sort option would be trivially easy; now that there's a dialog that presents options.. adding one more option wouldn't be a problem. On the one-hand, we want to avoid visual clutter. On the other hand, is this a feature that could be useful? If so.. I could certainly add it.

**Comment by ltguillaume** [comment]
I personally didn't really see the use, but I guess @Scumpeter could certainly still make their case again.
